src/CoreWCF.RabbitMQ.Client/src/CoreWCF/ServiceModel/Channels/RabbitMqChannelFactory.cs

```diff
@@ -0,0 +1,91 @@
+﻿using System.Linq;
+using System;
+using System.ServiceModel.Channels;
+using System.Threading.Tasks;
+
+namespace CoreWCF.ServiceModel.Channels
+{
+    public class RabbitMqChannelFactory : ChannelFactoryBase<IOutputChannel>
+    {
+        private BufferManager _bufferManager;
+        private MessageEncoderFactory _messageEncoderFactory;
+        private RabbitMqTransportBindingElement _transport;
+
+        internal RabbitMqChannelFactory(RabbitMqTransportBindingElement transport, BindingContext context)
+            : base(context.Binding)
+        {
+            _transport = transport;
+            _bufferManager = BufferManager.CreateBufferManager(transport.MaxBufferPoolSize, int.MaxValue);
+
+            var messageEncoderBindingElements = context.BindingParameters.OfType<MessageEncodingBindingElement>().ToList();
+            if (messageEncoderBindingElements.Count > 1)
+            {
+                throw new InvalidOperationException("More than one MessageEncodingBindingElement was found in the BindingParameters of the BindingContext");
+            }
+
+            if (messageEncoderBindingElements.Count == 1)
+            {
+                _messageEncoderFactory = messageEncoderBindingElements.First().CreateMessageEncoderFactory();
+            }
```

Comment:
Although this will go away with my previous comment, you did call `ToList()` so you don't need to use `First()`, you know there's a single item in there so could use `messageEncoderBindingElements[0]` instead.